common/djangoapps/config_models/models.py

```diff
@@ -35,33 +37,69 @@ def save(self, *args, **kwargs):
         Clear the cached value when saving a new configuration entry
         """
         super(ConfigurationModel, self).save(*args, **kwargs)
-        cache.delete(self.cache_key_name())
+        cache.delete(self.cache_key_name(*[getattr(self, key) for key in self.KEY_FIELDS]))
 
     @classmethod
-    def cache_key_name(cls):
+    def cache_key_name(cls, *args):
         """Return the name of the key to use to cache the current configuration"""
-        return 'configuration/{}/current'.format(cls.__name__)
+        if cls.KEY_FIELDS != ():
+            if len(args) != len(cls.KEY_FIELDS):
+                raise TypeError(
+                    "cache_key_name() takes exactly {} arguments ({} given)".format(len(cls.KEY_FIELDS), len(args))
+                )
+            return 'configuration/{}/current/{}'.format(
+                cls.__name__,
+                ','.join((arg if isinstance(arg, basestring) else repr(arg)) for arg in args)
```

Comment:
Well `repr` *is* better for use as a key because it's designed to be less ambiguous vs. `str` is designed for readability:
```python
>>> str(3)==str("3")
True
>>> repr(3)==repr("3")
False
```
But by using `str` sometimes and `repr` others I've perhaps somewhat defeated the purpose.

Maybe `repr(unicode(arg)) if isinstance(arg, str) else repr(arg)` ?